src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private MachineProducer<Truck> truckMachineProducer = new TruckProducer();
+    private MachineProducer<Excavator> excavatorMachineProducer = new ExcavatorProducer();
+    private MachineProducer<Bulldozer> bulldozerMachineProducer = new BulldozerProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        Map<String, MachineProducer<? extends Machine>> allMachines = new HashMap<>();
+        allMachines.put("Truck", truckMachineProducer);
+        allMachines.put("Excavator", excavatorMachineProducer);
+        allMachines.put("Bulldozer", bulldozerMachineProducer);
```

Comment:
use Map.of(), move it out of the method - no need for new map each time method is called